source/fundamentals/monitoring.txt

```diff
@@ -0,0 +1,409 @@
+==========
+Monitoring
+==========
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+.. What is on this page?
+
+This guide explains how you can set up and configure **monitoring** in the
+MongoDB Java driver.
+
+.. What do any new terms mean?
+
+Monitoring is the process of getting information about the activities a running 
+program performs for use in an application or an application performance
+management library. 
+
+Monitoring the MongoDB Java driver lets you understand the
+driver's resource usage and performance, and can help you make informed
+decisions when designing and debugging your application.    
+
+.. What can you expect to see on this page?
+
+In this guide you will learn how to perform these tasks:
+
+- :ref:`Monitor different types of events in the MongoDB Java Driver <monitoring-monitor-events>`
+- :ref:`Monitor connection pool events with Java Management Extensions (JMX) and JConsole <monitoring-jmx>`
+
+This guide shows how to use information about the activity of the driver in code.
+If you would like to learn how to record events the driver performs,
+consider reading our :doc:`guide on logging </fundamentals/logging>`.
+
+.. _monitoring-monitor-events:
+
+Monitor Events
+--------------
+
+To monitor an **event**, you must register a **listener** on your ``MongoClient``
+instance. 
+
+An event is any action that happens in a running program. The driver includes functionality
+for listening to a subset of the events that occur when the driver is running.
+
+A listener is a class that performs some action when certain events occur. All
+listener classes extend the ``EventListener`` interface. A listener's
```

Comment:
> All listener classes extend the ``EventListener`` interface

Not true in practice. The driver's event listener interfaces are not.

Reply:
Thanks for the note. Apologies, I didn't realize that this convention was not always followed in practice. I am removing content related to this from the page.

Purely for my own edification (feel free to ignore):

Why are the driver's events and listeners not subclasses of `EventListener` and `EventObject`? Just curious why this decision was made (no need to answer this if you are busy).

Reply:
I think those types go back to an early time, when listeners were used primarily in GUI applications.  They are just not useful in this context.